.github/workflows/deploy-tests.yml

```diff
@@ -16,7 +17,7 @@ jobs:
 
   detect-changelog-updates:
     needs: assess-file-changes
-    if: ${{ needs.assess-file-changes.outputs.SOURCE_CHANGED == 'true' }}
+    if: ${{ needs.assess-file-changes.outputs.SOURCE_CHANGED == 'true' && github.event.pull_request.draft == false }}
```

Comment:
@pauladkisson I think this is something good to be aware of. Before, in the second condition, I had the boolean with quotes:

`github.event.pull_request.draft == 'false'`

As we have in the first condition here (`needs.assess-file-changes.outputs.SOURCE_CHANGED == 'true'`) and in many other places. However, for some reason that did not work. I am wondering why those booleans behave differently. Could it be because it is ourselves who are setting them? Do you have any idea?

Now I changed the second condition boolean to false (without the quotes) and it works as intended. The changelog failed when I changed from draft to ready to review:

![image](https://github.com/user-attachments/assets/deef6a2a-ce5a-4251-a173-46ed879f0a26)

Reply:
I think it's because the inputs/outputs to/from reusable workflows and composite actions must be strings whereas the native github.event.pull_request.draft field is a proper boolean.

See https://github.com/actions/runner/issues/2238.

Reply:
Thanks for the pointer. I encountered this problem before in the cache action and I thought it would be good for you to be aware of it.

Reply:
This comment there in the thread you shared is also useful:

https://github.com/actions/runner/issues/2238#issuecomment-1686609260